File: JavaProject/src/com/enum1/SwitchEnumExample.java
package com.enum1;

enum Day1 {
    MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, SATURDAY, SUNDAY
}

public class SwitchEnumExample {
    public static void main(String[] args) {
        Day1 today = Day1.WEDNESDAY;

        // Using ordinal() to get the index of the current day
        System.out.println("Today is " + today + " and its ordinal value is " + today.ordinal());//return index

        // Using valueOf() to get the enum constant from a string
        Day1 dayFromString = Day1.valueOf("FRIDAY");//return same value 
        System.out.println("Day from string 'FRIDAY' is " + dayFromString);

        switch (today) {
            case MONDAY:
                System.out.println("Start of the work week!");
                break;
            case FRIDAY:
                System.out.println("Weekend is coming!");
                break;
            case SUNDAY:
                System.out.println("Rest day!");
                break;
            default:
                System.out.println("A regular day.");
        }
    }
}


//ordinal() at that particular index value  and values() is used return data valuesOf() to return value 

File: PythonProject/enum/SwitchEnumExample.py
from enum import Enum

# Define the Day1 enum
class Day1(Enum):
    MONDAY = 1
    TUESDAY = 2
    WEDNESDAY = 3
    THURSDAY = 4
    FRIDAY = 5
    SATURDAY = 6
    SUNDAY = 7

# Main logic using match-case (Python 3.10+)
if __name__ == "__main__":
    today = Day1.WEDNESDAY

    match today:
        case Day1.MONDAY:
            print("Start of the work week!")
        case Day1.FRIDAY:
            print("Weekend is coming!")
        case Day1.SUNDAY:
            print("Rest day!")
        case _:
            print("A regular day.")
